kasa/httpclient.py

```diff
@@ -49,6 +49,11 @@ async def post(
         response_data = None
         self._last_url = url
         self.client.cookie_jar.clear()
+        return_json = bool(json)
+        # If json is not a dict send as data
```

Comment:
When this shouldn't be the case? I mean, would it be better to this on the call site rather than hack around here?